cli/cmd/repair.go

```diff
@@ -43,6 +47,31 @@ linkerd upgrade.`,
 				return errors.New("secret/linkerd-config-overrides already exists. If you need to regenerate this resource, please delete it before proceeding")
 			}
 
+			// Check if the CLI version matches with that of the server
+			clientVersion := version.Version
+			var serverVersion string
+			publicClient, err := public.NewExternalClient(cmd.Context(), controlPlaneNamespace, k8sAPI)
+			if err != nil {
+				return err
+			}
+			serverVersion, err = healthcheck.GetServerVersion(cmd.Context(), publicClient)
+			if err != nil {
+				return err
+			}
+
+			if serverVersion != clientVersion {
```

Comment:
This is probably too strict, right?  There should be no incompatibility between 2.9.3 and 2.9.3, for example.  Or between most close edge versions.

Reply:
Yeah, It is strict. My thinking is that its impossible to know when we change config even btw edges and hence this way of doing it gets the correct new config always.

Any ideas on how to be a bit more relaxed without having to sacrifice correctness?

Reply:
Can we find the specific edge and stable versions where the breakage occurred?

Reply:
I don't think there is a direct way where we track what changes between versions in Values.yaml. 

We can see if the conversions actually error and fall back based on that but that would mean that we are compromising correctness i.e where some values got replaced but no error.